compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -736,14 +736,21 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
         fourthTry
     }
 
-    def tryBaseType(cls2: Symbol) = {
-      val base = nonExprBaseType(tp1, cls2)
-      if (base.exists && (base `ne` tp1))
-        isSubType(base, tp2, if (tp1.isRef(cls2)) approx else approx.addLow) ||
-        base.isInstanceOf[OrType] && fourthTry
-          // if base is a disjunction, this might have come from a tp1 type that
-          // expands to a match type. In this case, we should try to reduce the type
-          // and compare the redux. This is done in fourthTry
+    def tryBaseType(cls2: Symbol): Boolean = {
```

Comment:
This looks like it will overshoot. `tryBaseType` is an essential element for comparisons. Simply ignoring this step under GADT constraint inference is too drastic. I would expect instead a necessaryEither somewhere in the logic.

Reply:
We only avoid `tryBaseType` if the LHS is an AndType. In that case, calculating the base type can end up performing a comparison that is too approximate to be useful for GADTs (see neg/i11545a.scala, added in this PR). I have ideas for other fixes, we can discuss them during today's lab meeting.